packages/webcert/src/feature/certificate/Modals/SignCertificateModal.tsx

```diff
@@ -0,0 +1,227 @@
+          </p>
+          <p>
+            Det verkar som att du valt en annan identitet att signera med än den du loggade in med. Du måste identifiera dig på samma sätt
+            som när du loggade in. Kontrollera om du har valt rätt och prova igen.
+          </p>
+        </>
+      )
+    case ErrorCode.INTYG_FROM_OTHER_VARDGIVARE_EXISTS: {
+      switch (certificateType) {
+        case 'db':
+          return (
+            <p>
+              Dödsbevis för detta personnummer har utfärdats av någon annan hos annan vårdgivare under tiden du har arbetat med detta
+              utkast. Det är inte möjligt att signera detta dödsbevis.
+            </p>
+          )
+        case 'doi':
+          return (
+            <p>
+              Dödsorsaksintyg för detta personnummer har utfärdats av någon annan hos annan vårdgivare under tiden du har arbetat med detta
+              utkast. Senast skapade dödsorsaksintyg är det som gäller. Om du fortsätter och lämnar in dödsorsaksintyget så blir det därför
+              detta dödsorsaksintyg som gäller.
+            </p>
+          )
+      }
+      break
+    }
+    case ErrorCode.INTYG_FROM_SAME_VARDGIVARE_EXISTS: {
+      switch (certificateType) {
+        case 'db':
+          return (
+            <p>
+              Dödsbevis för detta personnummer har utfärdats av någon annan under tiden du har arbetat med detta utkast. Du kan inte signera
+              detta dödsbevis men kan däremot välja att ersätta det befintliga dödsbeviset.
+            </p>
+          )
+        case 'doi':
+          return (
+            <p>
+              Dödsorsaksintyg för detta personnummer har utfärdats av någon annan under tiden du har arbetat med detta utkast. Du kan inte
+              signera detta dödsorsaksintyg men kan däremot välja att ersätta det befintliga dödsorsaksintyget.
+            </p>
+          )
+      }
+      break
+    }
+    case ErrorCode.GRP_PROBLEM: {
+      switch (message) {
+        case 'ALREADY_IN_PROGRESS':
+          return (
+            <p>
+              <strong>Intyget kunde inte signeras.</strong>
+              <br />
+              En inloggning eller underskrift för det här personnumret är redan påbörjad, tryck avbryt i BankID säkerhetsapp och försök
+              igen.
+            </p>
+          )
+        case 'USER_CANCEL':
+        case 'CANCELLED':
+          return (
+            <p>
+              <strong>Intyget kunde inte signeras.</strong>
+              <br />
+              Åtgärden avbruten.
+            </p>
+          )
+      }
+    }
+  }
+
+  return (
+    <p>
+      Intyget har inte signerats. Detta beror antingen på ett tekniskt fel eller att signeringen avbrutits. <br />
+      Prova igen om en stund. Om problemet kvarstår, kontakta i första hand din lokala IT-avdelning och i andra hand{' '}
+      <WCDynamicLink linkKey="ineraKundserviceAnmalFel" />.
+    </p>
+  )
+}
+
+export const SignCertificateModal: React.FC = () => {
```

Comment:
I think we need tests for this modal :)

Reply:
Good call, I'll add a couple of tests for the modal

Reply:
unit tests have been added.